cpp/tests/strings/reverse_tests.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+
+#include <cudf/column/column.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/strings/reverse.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+#include <vector>
+
+struct StringsReverseTest : public cudf::test::BaseFixture {
+};
+
+TEST_F(StringsReverseTest, Reverse)
+{
+  auto input = cudf::test::strings_column_wrapper(
+    {"abcdef", "12345", "", "", "aébé", "A é Z", "X", "é"}, {1, 1, 1, 0, 1, 1, 1, 1});
+  auto results  = cudf::strings::reverse(cudf::strings_column_view(input));
+  auto expected = cudf::test::strings_column_wrapper(
+    {"fedcba", "54321", "", "", "ébéa", "Z é A", "X", "é"}, {1, 1, 1, 0, 1, 1, 1, 1});
```

Comment:
Are any of these multiple byte characters? Should we have multiple byte characters in the test? I don't recall if we support them or not.

Reply:
All of the accented ones (`é`) are multi-byte UTF-8 characters.